### ISSUE TITLE
Is a detector needed?

### ISSUE THREAD

**Issue by HonWek** [zju3dv/clean-pvnet#208]
Hi,

Thank you for making this work open-source. Currently, I am using PVNet for pose estimation in a Gazebo simulation environment by running the inference as a node in ROS. I have achieved good training results after more than 200 epochs of training. Using visualization, I am able to see that there is a good overlap between the ground truth bounding box and predicted bounding box. However, when I run the inference for pose estimation in real-time with Gazebo, the results are not good at all, and it seems that the object cannot even be detected.

While I am searching for the possible causes, I found that you mentioned that a detector is needed to run the inference in real-time. However, I am kind of confused as I do not see where the detector comes in while reading the codes. Specifically, you mentioned that a detector is needed at #6. But checking the codes that you provided at #125 and the codes that another person provided at #109, they do not seem to require a detector. According to your reply on another repo at [here](https://github.com/zju3dv/pvnet/issues/63#issuecomment-521844903), the inference can be run directly using images received from the camera. In addition, with the demo codes that you provided [here](https://github.com/zju3dv/clean-pvnet/blob/5870c509e3cc205e1bb28910a7b1a9a3c8add9a8/run.py#L246), I also do not see the use of a detector.

Hence, may I know whether a detector is needed to run the inference? And specifically,
1. What are the preprocessing steps of the images needed before I feed them into the network with the line `output = network(inp)` , apart from converting them to PyTorch tensor and normalizing them?
2. Could you recommend a suitable detector, if one is needed? Would it be possible to provide a sample code if you have used a detector in conjunction with this PVNet before?

I look forward to hearing from you, and any help would be very much appreciated. Thank you in advance!

**Comment by pengsida** [comment]
The detector is optional.

**Comment by pengsida** [comment]
We have provided a detector in thie repo.

You could track the code by running `python run.py --type visualize --cfg_file configs/tless/tless_01.yaml test.det_gt True`.
